aws/table_aws_cost_by_tag.go

```diff
@@ -0,0 +1,130 @@
+package aws
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/costexplorer"
+	"github.com/aws/aws-sdk-go-v2/service/costexplorer/types"
+	goKitTypes "github.com/turbot/go-kit/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+func tableAwsCostAndUsageByTag(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_cost_by_tags",
+		Description: "AWS Cost Explorer - Cost and Usage By Tags",
+		List: &plugin.ListConfig{
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "granularity", Require: plugin.Required},
+				{Name: "tag_key_1", Require: plugin.Required},
+				{Name: "tag_key_2", Operators: []string{"=", "<>"}, Require: plugin.Optional, CacheMatch: "exact"},
+			},
+			Hydrate: listCostAndUsageByTags,
+		},
+		Columns: awsColumns(
+			costExplorerColumns([]*plugin.Column{
+
+				// Quals columns - to filter the lookups
+				{
+					Name:        "granularity",
+					Description: "",
+					Type:        proto.ColumnType_STRING,
+					Hydrate:     hydrateCostAndUsageQuals,
+				},
+				{
+					Name:        "tag_key_1",
+					Description: "The tag key to group by.",
+					Type:        proto.ColumnType_STRING,
+					Hydrate:     hydrateCostAndUsageQuals,
+				},
+				{
+					Name:        "tag_key_2",
+					Description: "A secondary tag key to group by.",
+					Type:        proto.ColumnType_STRING,
+					Hydrate:     hydrateCostAndUsageQuals,
+				},
+				{
```

Comment:
I think we should put the tag value columns immediately after the corresponding tag key column, so that when you do `select *...` you see them next to each other (granularity, tag_key_1, tag_value_1, tag_key_2, tag_value_2)

Reply:
Updated the column ordering.